### PR TITLE
Add the docs for supporting event watcher in app config

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ spec:
 | notification | [DeploymentNotification](#deploymentnotification) | Additional configuration used while sending notification to external services. | No |
 | postSync | [PostSync](#postsync) | Additional configuration used as extra actions once the deployment is triggered. | No |
 | variantLabel | [KubernetesVariantLabel](#kubernetesvariantlabel) | The label will be configured to variant manifests used to distinguish them. | No |
+| eventWatcher | [][EventWatcher](#eventWatcher) | Configuration for event watcher. | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:40]: should be `List of ...` or such to mention that this is in plural form? 🤔

File: docs/content/en/docs-dev/user-guide/configuration-reference.md [hunk 2]
```diff
@@ -64,6 +65,7 @@ spec:
 | timeout | duration | The maximum length of time to execute deployment before giving up. Default is 6h. | No |
 | notification | [DeploymentNotification](#deploymentnotification) | Additional configuration used while sending notification to external services. | No |
 | postSync | [PostSync](#postsync) | Additional configuration used as extra actions once the deployment is triggered. | No |
+| eventWatcher | [][EventWatcher](#eventWatcher) | Configuration for event watcher. | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/configuration-reference.md:68]: should be same as L40 and other as well 👀

Comment (knanao) [reply docs/content/en/docs-dev/user-guide/configuration-reference.md]: Right, thank you.

**Review by khanhtc1202 — APPROVED** [review]
👍

**Review by nghialv — APPROVED** [review]
Thank you.

**Pull request merged**